network-mux/src/Network/Mux/Bearer/Socket.hs

```diff
@@ -73,10 +74,10 @@ socketAsMuxBearer tracer sd = do
                   return (header {Mx.msBlob = blob}, ts)
 
       recvLen' :: Bool -> Int64 -> [BL.ByteString] -> IO BL.ByteString
-      recvLen' _ 0 bufs = return (BL.concat $ reverse bufs)
-      recvLen' waitingOnNxtHeader l bufs = do
+      recvLen' _ 0 !bufs = return (BL.concat $ reverse bufs)
+      recvLen' waitingOnNxtHeader l !bufs = do
           traceWith tracer $ Mx.MuxTraceRecvStart $ fromIntegral l
-          buf <- Socket.recv sd l
+          !buf <- Socket.recv sd l
```

Comment:
Wouldn't it be better to ensure `Socket.recv` return a result that is in normal form?

Reply:
I think we only can have an implication `WHNF => NF` for values returned by a lambda. This bang will ensure that it is in WHNF. A test for `WHNF => NF` would be useful.